src/main/java/Duke.java

```diff
@@ -1,10 +1,130 @@
+import java.util.Scanner;
+import java.util.Arrays;
+import java.lang.String;
+
 public class Duke {
+    public static Task[] taskList = new Task[100];
+    public static int taskCount = 0;
+    public static final String LINE_HEADER = "\n=====================================================\n";
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        String inputCommand;
+        printGreeting();
+
+        // Command handler
+        do {
+            Scanner in = new Scanner(System.in);
+            inputCommand = in.nextLine();
+
+            String taskObj = new Scanner(inputCommand).next();
+
+            switch (taskObj){
+                case "list":
+                    listAllTasks();
+                    break;
+
+                case "bye":
+                    printGoodbye();
+                    break;
+
+                case "done":
+                    String[] commandWords = inputCommand.split(" ");
+                    int taskNumber = Integer.parseInt(commandWords[1])-1;
+                    markTaskAsDone(taskNumber);
+                    break;
+
+                case "todo":
+                    String taskDetails = inputCommand.substring(4);
+                    addTask(new ToDo(taskDetails));
```

Comment:
This still involve lower level details, the only part that is being abstracted is adding the ToDo to the task list. 
Should abstract the creation of the ToDo object and the splicing up of the inputCommand.